### ISSUE TITLE
[wgsl-in] panic on return without location

### ISSUE THREAD

**Issue by SNCPlay42** [gfx-rs/naga#807]
This code:

```
[[stage(vertex)]]
fn main() -> f32 {
    return 0.0;
}
```

should error because the return has no location, but produces this panic, using naga 0.4.0:

```
  thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', \.cargo\registry\src\github.com-1ecc6299db9ec823\naga-0.4.0\src\proc\interpolator.rs:90:52  
  stack backtrace:
     0: std::panicking::begin_panic_handler
               at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0\/library\std\src\panicking.rs:493
     1: core::panicking::panic_fmt
               at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0\/library\core\src\panicking.rs:92
     2: core::panicking::panic
               at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0\/library\core\src\panicking.rs:50
     3: core::option::Option<mut naga::Binding*>::unwrap<mut naga::Binding*>
               at \.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\option.rs:386
     4: naga::proc::interpolator::{{impl}}::apply_common_default_interpolation::default_binding_or_struct
               at \.cargo\registry\src\github.com-1ecc6299db9ec823\naga-0.4.0\src\proc\interpolator.rs:90
     5: naga::Module::apply_common_default_interpolation
               at \.cargo\registry\src\github.com-1ecc6299db9ec823\naga-0.4.0\src\proc\interpolator.rs:135
     6: naga::front::wgsl::Parser::parse
               at \.cargo\registry\src\github.com-1ecc6299db9ec823\naga-0.4.0\src\front\wgsl\mod.rs:2745
     7: naga::front::wgsl::parse_str
               at \.cargo\registry\src\github.com-1ecc6299db9ec823\naga-0.4.0\src\front\wgsl\mod.rs:2754
     8: build_script_build::main
               at .\build.rs:24
     9: core::ops::function::FnOnce::call_once<fn(),tuple<>>
               at \.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
```

If the return value is given a location attribute, parsing succeeds as expected:

```
// no panic
[[stage(vertex)]]
fn main() -> [[location(0)]] f32 {
    return 0.0;
}
```

This also applies to fields on returned structs without a location:

```
// also panics
struct Output {
    [[location(0)]] ok: f32;
    bad: f32;
};

[[stage(vertex)]]
fn main() -> Output {
    var output: Output;
    output.ok = 0.0;
    output.bad = 0.0;
    return output;
}
```

**Comment by kvark** [comment]
@jimblandy would you be able to address this? The interpolator should gracefully error instead of panicking, since by the time we call it there was no validation yet.

**Comment by kvark** [comment]
Likely fixed by #894
